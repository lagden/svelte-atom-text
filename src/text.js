/* istanbul ignore file */
import Text from './Text.svelte'

export {
	Text
}
