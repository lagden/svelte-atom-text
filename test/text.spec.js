/* globals describe, afterEach, test, expect */

import {cleanup, render} from '@testing-library/svelte'
import {Text} from '../src/text.js'

describe('Text', () => {
	afterEach(cleanup)

	test('should match snapshot default', () => {
		const {container} = render(Text, {
			props: {
				id: 'foo',
				name: 'test',
				outline: false,
				showError: false
			}
		})
		expect(container).toMatchSnapshot()
	})

	test('should match snapshot props', () => {
		const {container} = render(Text, {
			props: {
				className: 'customCssDiv',
				class: 'customCss',
				type: 'tel',
				required: true,
				id: 'bar',
				name: 'test',
				label: '<b>Test</b>'
			}
		})
		expect(container).toMatchSnapshot()
	})
})
