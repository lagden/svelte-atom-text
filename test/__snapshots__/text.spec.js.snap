// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should match snapshot default 1`] = `
<body>
  <div>
    <div
      class=""
    >
       
      <input
        class="_atom_frm__base _atom_frm__standard "
        id="foo"
        name="test"
        type="text"
      />
       
    </div>
  </div>
</body>
`;

exports[`Text should match snapshot props 1`] = `
<body>
  <div>
    <div
      class="customCssDiv"
    >
      <label
        class="_atom_frm__label _atom_frm__label___required"
        for="bar"
        required="true"
      >
        <b>
          Test
        </b>
        
      </label>
       
      <input
        class="_atom_frm__base _atom_frm__outline customCss"
        id="bar"
        name="test"
        required=""
        type="tel"
      />
       
      <small
        class="_atom_frm__error___message"
        title="Constraints not satisfied"
      >
        Constraints not satisfied
      </small>
    </div>
  </div>
</body>
`;
